spec/game_runner_spec.rb

```diff
@@ -26,12 +27,29 @@
 
       it "prompts a user for input" do
         stdout = StringIO.new
-        game = GameRunner.new(stdout)
+        stdin = StringIO.new
+        game = GameRunner.new(stdout, stdin)
 
         game.run
 
         expect(stdout.string).to include("Enter your move >")
       end
+
+      it "returns a message if the input is invalid, and exits when given valid input" do
+        stdout = StringIO.new
+        stdin = StringIO.new
+        game = GameRunner.new(stdout, stdin)
+        allow(stdin).to receive(:gets).and_return("Z6", "B12", "A3")
+
+        game.run
+
```

Comment:
This series of `expect` calls is not working the way you are thinking. You already stubbed the response for `stdin.gets`. So your first expectation below becomes in english:

I expect stdin to receive `gets`, once and return nothing. The next time, the expectation is the same, and then the third time.

With the string, you are expecting the string to include the same thing over and over again. If it include that text once though, it will include it the next time. Because your expect does not trigger a `gets`. Those `gets` happen all at once when the game is run. 

You can't expect the game to exit with the `exit` keyword ... it will just do it.